sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -41,7 +41,20 @@ object CatalystSerde {
   }
 
   def generateObjAttr[T : Encoder]: Attribute = {
-    AttributeReference("obj", encoderFor[T].deserializer.dataType, nullable = false)()
+    val deserializer = encoderFor[T].deserializer
+    val dataType = deserializer.dataType
+    val nullable = if (deserializer.childrenResolved) {
```

Comment:
> Except for the deserializer doesn't refer any attribute, otherwise a resolved deserializer can't pass the check, I think.

The only possible that `encoderFor[T].deserializer.resolved` is true, is the `deserializer` doesn't refer anything. I think it is meaningless and abnormal query like the test case.

If I don't miss anything, the value of returned `nullable` shouldn't matter.